cloud_experts_tutorials/cloud-experts-deploying-application/cloud-experts-deploying-application-integrating-aws.adoc

```diff
@@ -0,0 +1,394 @@
+
+Pods receive a valid OpenID Connect (OIDC) JSON web token (JWT) and pass it to the AWS STS `AssumeRoleWithWebIdentity` API operation to receive IAM temporary role credentials. The process relies on the EKS pod identity mutating webhook which modifies pods that require AWS IAM access.
+
+IAM roles for service accounts adheres to the following best practices:
+
+* *Principle of least privilege*: You can create IAM permissions for AWS roles that only allow limited access. These permissions are limited to the service account associated with the role and only the pods that use that service account have access.
+* *Credential isolation*: A pod can only retrieve credentials for the IAM role associated with the service account that the pod is using.
+* *Auditing*: All AWS resource access can be viewed in CloudTrail.
+
+[role="_additional-resources"]
+.Additional resources
+
+* link:https://aws-controllers-k8s.github.io/community/docs/user-docs/irsa/[ACK documentation page]
+
+== Installing the ACK controller
+Install the ACK controller to create and delete buckets in the S3 service by using a Kubernetes custom resource for the bucket. Installing the controller will also create the required namespace and service account.
+
+We will use an Operator to make it easy. The Operator installation will also create an `ack-system` namespace and a service account `ack-s3-controller` for you.
+
+. Log in to the cluster console.
+. On the left menu, click *Operators*, then *OperatorHub*.
+. In the filter box, enter "S3" and select *AWS Controller for Kubernetes - Amazon S3*.
++
+image:cloud-experts-deploying-integrating-ack-operator.png[]
++
+. If a pop-up about community operators appears, click *Continue*.
+. Click *Install*.
+. Select *All namespaces on the cluster* under "Installation mode".
+. Select *ack-system* under "Installed Namespace".
+. Select *Manual* under "Update approval".
++
+[IMPORTANT]
+====
+Make sure *Manual Mode* is selected so changes to the service account are not overwritten by an automatic operator update.
+====
++
+. Click *Install*.
++
+The settings should look like the below image.
++
+image:cloud-experts-deployment-integrating-ack-install.png[]
++
+. Click *Approve*.
+. The installation begins but will not complete until you have created an IAM role and policy for the ACK controller.
+
+== Creating an IAM role and policy for the ACK controller
+
+. Run one of the following scripts to create the AWS IAM role for the ACK controller and assign the S3 policy:
+** Use the link:https://www.rosaworkshop.io/ostoy/resources/setup-s3-ack-controller.sh[setup-s3-ack-controller.sh] script, which automates the process for you.
```

Comment:
Not sure if you need to add a note, but it was a little startling that this just downloaded a file for me. Probably fine though.